### PR TITLE
fix: `watchConnections` emits a disconnect even during a page refresh

### PR DESCRIPTION
**Pull request by magiziz**
# Description

A very unusual edge case, but when refreshing the page after connecting via email or social login a disconnection occurs before reconnecting. Instead of relying on `watchConnections` which causes this issue we should use `watchAccount` only for emitting disconnect event.

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

For Linear issues: Closes APKT-2280

# Checklist

- [x] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [x] My changes generate no new warnings
- [x] I have reviewed my own code
- [x] I have filled out all required sections
- [ ] I have tested my changes on the preview link
- [ ] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

File: packages/adapters/wagmi/src/client.ts [hunk 1]
```diff
@@ -240,13 +240,6 @@ export class WagmiAdapter extends AdapterBlueprint {
         }
       }
     })
-    watchConnections(this.wagmiConfig, {
-      onChange: connections => {
-        if (connections.length === 0) {
-          this.emit('disconnect')
-        }
-      }
-    })
```
Comment (magiziz) [inline packages/adapters/wagmi/src/client.ts:249]: We already have tests for `watchAccount`

**Review by enesozturk — APPROVED** [review]

**Comment by enesozturk** [comment]
@magiziz There are some ts issues

**Review by tomiir — APPROVED** [review]
Can we add a test?

**Comment by magiziz** [comment]
@tomiir sorry might have resolved a comment i put, but seems like we already have test covered for `watchAccount` https://github.com/reown-com/appkit/pull/3970#discussion_r1984846734

File: packages/adapters/wagmi/src/client.ts (hunk 1 above)
Comment (magiziz) [reply packages/adapters/wagmi/src/client.ts:249]: It only happens when you're logged in with social provider and refresh the page. The wagmi thinks there are no connections at first and then it emits disconnect event.

I think it's only possible to catch this in an e2e test, but it's an extreme edge case

Comment (magiziz) [reply packages/adapters/wagmi/src/client.ts:249]: i was only able to reproduce this in web wallet

Comment (svenvoskamp) [reply packages/adapters/wagmi/src/client.ts:249]: I remember I added this for a specific case, but can't remember anymore. Not sure if it save to delete

Comment (svenvoskamp) [reply packages/adapters/wagmi/src/client.ts:249]: But this was before we did improvements on initialisation AppKit, so might be okay to remove it. We just need make sure this gets manually tested
